File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class Main {
    public static void main(String[] args) throws IOException {
        // Initialize the buffered reader and writer for input and output
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));

        // Read the first input value n
        int n = Integer.parseInt(reader.readLine());

        // Read the first array of integers inputN
        String[] inputN = reader.readLine().split(" ");
        // Read the second input value m
        int m = Integer.parseInt(reader.readLine());

        // Read the second array of integers inputM
        String[] inputM = reader.readLine().split(" ");

        // Create a map to store the count of each number in inputN
        Map<Integer, Integer> hashMap = new HashMap<>();

        // For each number in inputN, update its count in the map
        for (int i = 0; i < n; i++) {
            int num = Integer.parseInt(inputN[i]);
            hashMap.put(num, hashMap.getOrDefault(num, 0) + 1);
        }

        // For each number in inputM, write the count from hashMap or 0 if not present
        for (int i = 0; i < m; i++) {
            int num = Integer.parseInt(inputM[i]);
            writer.write(hashMap.getOrDefault(num, 0) + " ");
        }

        // Flush and close the writer
        writer.flush();
        writer.close();
    }
}

File: 백준/Silver/10816. 숫자 카드 2/숫자 카드 2.kt
import java.util.*
import kotlin.collections.ArrayList
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val n = readLine().toInt()

    val inputN = readLine().split(" ").map {it.toInt()}

    val m = readLine().toInt()

    val inputM = readLine().split(" ").map {it.toInt()}

    val hashMap = HashMap<Int,Int>() //카운트를 담는 Map

    repeat(n) {
        //첫번째 입력인 n 만큼 반복하여 만약 배열의 숫자가 여러개라면 1씩 누적시키는 코드입니다.
        //getOrDefault로 처음 hashMap에 들어갈 때, 0으로 들어가도록 해줬습니다.
        hashMap[inputN[it]] = hashMap.getOrDefault(inputN[it],0) + 1
    }

    repeat(m) {
        bw.write("${hashMap[inputM[it]] ?: 0} ")
    }

    bw.flush()
    bw.close()
}